CHANGELOG.next.toml

```diff
@@ -11,6 +11,12 @@
 # meta = { "breaking" = false, "tada" = false, "bug" = false }
 # author = "rcoh"
 
+[[smithy-rs]]
+message = "Switch to `base64` crate"
+references = ["smithy-rs#1294"]
+meta = { "breaking" = true, "tada" = false, "bug" = false }
```

Comment:
Where do I add this message?

> The base64 implementation from `aws-smithy-types` has been removed. Please switch to another implementation (e.g. the [`base64`](https://crates.io/crates/base64) crate).

I wanted to add it to `aws/SDK_CHANGELOG.md`, but that file is autogenerated from `update-changelog` from `tools/sdk-lints`. How are the migration guides from that file added if the file is autogenerated? Do they get added manually just before preparing a release?

Reply:
if you see the comments above, you use `[[aws-sdk-rust]]` instead of `smithy-rs`.

Anything you want in the message goes in the message field—TOML allows multiline strings. **However**: I think we should keep the same function signatures and just delegate to the base64 crate behind the scenes. Having a "blessed" base64 implementation inside of `aws-smithy-types` is probably generally a good thing to have.

Reply:
> There's an example above—[[aws-sdk-rust]] instead of smithy-rs.

But how do I add a longer migration guide e.g. https://github.com/awslabs/smithy-rs/blob/775d227c5fbc46c2e3c2878b270b8e8226ab7ef1/aws/SDK_CHANGELOG.md?plain=1#L5-L33 ? Do I plop all of that Markdown into `message`?

---

>  However: I think we should keep the same function signatures and just delegate to the base64 crate behind the scenes. Having a "blessed" base64 implementation inside of aws-smithy-types is probably generally a good thing to have.

I thought a bit about this before making the change and concluded that there's no benefit. base64 is standard and I can't foresee us or Smithy wanting to customize anything that is not already customizable via the `base64` crate. I also don't think that SDK users should have public access to a base64-encoding routine from a runtime crate; they should have no need for it after all, since they should just be using `aws_smithy_types::Blob`. I think having a "blessed" base64 implementation is unnecessarily increasing public API surface.

Reply:
This makes sense!

Reply:
Maybe blob should add a from_base64 method...

Reply:
yeah, just use triple quote strings for long messages

Reply:
Ok, I've added it here https://github.com/awslabs/smithy-rs/pull/1294/commits/f1c00cbd41c05c5be8dab3bc5392c4583a6ae35d